tests/AbstractLoopTest.php

```diff
@@ -16,11 +16,39 @@ public function setUp()
 
     abstract public function createLoop();
 
+    /**
+     * @deprecated Use createSocketPair() instead.
+     */
     public function createStream()
     {
         return fopen('php://temp', 'r+');
     }
+    
+    public function isUnixSocketSupported()
+    {
+        return in_array('unix', stream_get_transports(), true);
+    }
 
+    public function createSocketPair()
+    {
+        if ($this->isUnixSocketSupported()) {
+            $sockets = stream_socket_pair(STREAM_PF_UNIX, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
+        } else {
+            $sockets = stream_socket_pair(STREAM_PF_INET, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
+        }
+        
+        foreach ($sockets as $socket) {
+            if (function_exists('stream_set_read_buffer')) {
```

Comment:
This check could be performed before the loop. However, this is lacking some documentation, why is this needed in the first place? And why do we not bother if it doesn't exist?